### PR TITLE
[NPM-3356] Improve retries of DNSStats tests

### PR DESCRIPTION
**Pull request by akarpz**
### What does this PR do?

Updates the helper for TestDNSStats* unit tests to allow retries to occur, and better collect errors.

### Motivation

https://datadoghq.atlassian.net/browse/NPM-3356

### CODE REVIEW

**Comment by akarpz** [comment]
unfortunately even with this change I still saw this in the test: 
```
=== RUN   TestTracerSuite/CO-RE/TestDNSStatsWithNAT
    tracer_linux_test.go:1398: requesting golang.com@2.2.2.2 with conntrack type: *tracer.ebpfConntracker
    tracer_test.go:1005:
        	Error Trace:	/git/datadog-agent/pkg/network/tracer/tracer_test.go:1005
        	            				/git/datadog-agent/pkg/network/tracer/tracer_linux_test.go:1399
        	Error:      	Not equal:
        	            	expected: 0x1
        	            	actual  : 0x0
        	Test:       	TestTracerSuite/CO-RE/TestDNSStatsWithNAT
        	Messages:   	expected 1 successful responses but got 0
```

File: pkg/network/tracer/tracer_test.go [hunk 1]
```diff
@@ -960,47 +960,51 @@ func testDNSStats(t *testing.T, tr *Tracer, domain string, success, failure, tim
...
-	var dnsClientAddr *net.UDPAddr
-	require.Eventually(t, func() bool {
+	// we place the entire test in the eventually loop since failed DNS requests can cause "WaitForDomain" to return
+	// true before DNS data is actually available for the successful connection
+	require.EventuallyWithT(t, func(c *assert.CollectT) {
 		dnsClient := new(dns.Client)
 		dnsConn, err := dnsClient.Dial(dnsServerAddr.String())
-		require.NoError(t, err)
-		dnsClientAddr = dnsConn.LocalAddr().(*net.UDPAddr)
+		assert.NoError(c, err)
```
Comment (hmahmood) [inline pkg/network/tracer/tracer_test.go:968]: Should we continue if any of these `assert`'s fire? All of these should return `false` if they fire, so we can return early in that case.

Comment (akarpz) [reply pkg/network/tracer/tracer_test.go:968]: yeah that's correct. I think that's what we were doing before, just not handling the race

**Review by hmahmood — APPROVED** [review]

**Pull request merged**